go.mod

```diff
@@ -198,6 +204,8 @@ require (
 	github.com/tv42/httpunix v0.0.0-20191220191345-2ba4b9c3382c // indirect
 	github.com/vmware/govmomi v0.18.0 // indirect
 	go.etcd.io/bbolt v1.3.6 // indirect
+	go.mozilla.org/gopgagent v0.0.0-20170926210634-4d7ea76ff71a // indirect
+	go.mozilla.org/sops/v3 v3.7.1 // indirect
```

Comment:
Just a small question, any idea why this shows up as an `indirect` dependency? I see the config code directly imports this module so I'd expect it to appear like the Vault libs, cloud SDKs, etc. appear in the go.mod list. Not a big deal necessarily but I'd like to be sure about it so we don't get weird behavior with dependency updates, etc.

Reply:
I have been looking into this and `go mod tidy` does not appear to remove indirect without additional work.

Tidy currently reports few instances of

> but go 1.16 would select

If I run the recommended tidy command to fix it then indirect is removed from sops but that also changes many other dependancies and should therefore be a different PR

Reply:
I checked out your fork (`-b sops-backend`) and ran `go mod tidy -compat=1.17` and got this output:

```
Joshuas-MacBook-Pro-2:argocd-vault-plugin jkayani$ git diff go.mod 
diff --git a/go.mod b/go.mod
index 611e6ad..a291f07 100644
--- a/go.mod
+++ b/go.mod
@@ -43,6 +43,7 @@ require (
        github.com/spf13/cobra v1.3.0
        github.com/spf13/viper v1.10.1
        go.mongodb.org/mongo-driver v1.8.1 // indirect
+       go.mozilla.org/sops/v3 v3.7.1
        golang.org/x/crypto v0.0.0-20211108221036-ceb1ce70b4fa // indirect
        golang.org/x/net v0.0.0-20211216030914-fe4d6282115f
        golang.org/x/oauth2 v0.0.0-20211104180415-d3ed0bb246c8 // indirect
@@ -205,7 +206,6 @@ require (
        github.com/vmware/govmomi v0.18.0 // indirect
        go.etcd.io/bbolt v1.3.6 // indirect
        go.mozilla.org/gopgagent v0.0.0-20170926210634-4d7ea76ff71a // indirect
-       go.mozilla.org/sops/v3 v3.7.1 // indirect
        go.opencensus.io v0.23.0 // indirect
        go.uber.org/atomic v1.9.0 // indirect
        golang.org/x/sync v0.0.0-20210220032951-036812b2e83c // indirect
```

I think this is all I'd like to see - can you confirm this works the same way for you (it should, right?) and commit/push this if so?

Reply:
Did you also look at the change this makes to `go.sum`?

If you are comfortable with it then I will make this change

Reply:
I did - I'm not too concerned about them since none of the versions in `go.mod` are changing. Thanks for checking though

Reply:
The code has been updated with this change